### PR TITLE
test: Tests and Storybook entry for the Popover component

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
Adds tests for the Popover component and a storybook entry with controls.

### CODE REVIEW

**Comment by geido** [comment]
@junlincc @rusackas

File: superset-frontend/src/components/Popover/Popover.test.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+test('it should not render a title or content when not visible', () => {
+  render(<Popover content="Content sample" title="Popover title" />);
+  const content = screen.queryByText('Content sample');
+  const title = screen.queryByText('Popover title');
+  expect(content).not.toBeInTheDocument();
+  expect(title).not.toBeInTheDocument();
+});
+
+test('renders with icon child', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/components/Popover/Popover.test.tsx:50]: Don't need to be async

File: superset-frontend/src/components/Popover/Popover.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import React from 'react';
+import Popover from '.';
+
+export default {
+  title: 'Popover',
+  component: Popover,
+};
+
+export const InteractivePopover = (
+  args: JSX.IntrinsicAttributes & PopoverProps & React.RefAttributes<unknown>,
```
Comment (michael-s-molina) [inline superset-frontend/src/components/Popover/Popover.stories.tsx:30]: Do you need these intersections? Shouldn't `PopoverProps` be sufficient?

**Review by michael-s-molina — COMMENTED** [review]
Very cool @geido!

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Review by rusackas — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**